clients/v1_repl/v1_repl.go

```diff
@@ -0,0 +1,912 @@
+func reSubMatchMap(r *regexp.Regexp, str string) *map[string]string {
+	match := r.FindStringSubmatch(str)
+	if match == nil {
+		return nil
+	}
+	subMatchMap := make(map[string]string)
+	for i, name := range r.SubexpNames() {
+		if i != 0 {
+			subMatchMap[name] = match[i]
+		}
+	}
+	return &subMatchMap
+}
+
+var insertIntoRegex string = `^(?i)INSERT INTO ` + identRegex("db") + `(\.` + identRegex("rp") + `)? (?P<point>.+)$`
+var insertRegex string = `^(?i)INSERT (?P<point>.+)$`
+
+func (c Client) insert(cmd string) {
+	var db string
+	var rp string
+	var point string
+	insertRgx := regexp.MustCompile(insertRegex)
+	insertIntoRgx := regexp.MustCompile(insertIntoRegex)
+	insertMatches := reSubMatchMap(insertRgx, cmd)
+	insertIntoMatches := reSubMatchMap(insertIntoRgx, cmd)
+	if insertIntoMatches != nil {
+		db = getIdentFromMatches(insertIntoMatches, "db")
+		rp = getIdentFromMatches(insertIntoMatches, "rp")
+		point = getIdentFromMatches(insertIntoMatches, "point")
+		if db != "" && rp == "" {
+			defaultRp, err := c.getDefaultRetentionPolicy(context.Background(), db)
+			if err != nil {
+				color.Red("Unable to get default retention policy for %q", db)
+				return
+			}
+			rp = defaultRp
+		}
+	} else if !strings.HasPrefix(strings.ToUpper(cmd), "INSERT INTO") && insertMatches != nil {
+		db = c.Database
+		rp = c.RetentionPolicy
+		point = getIdentFromMatches(insertMatches, "point")
+		if db == "" {
+			color.Red("Please run \"use <database>\" to run \"INSERT <point\"")
+			return
+		}
+	} else {
+		color.Red("Expected \"INSERT INTO <database>.<retention_policy> <point>\" OR \"INSERT <point>\".")
+		return
+	}
+
+	buf := bytes.Buffer{}
+	gzw := gzip.NewWriter(&buf)
+
+	_, err := gzw.Write([]byte(point))
+	gzw.Close()
+	if err != nil {
+		color.Red("Failed to gzip points")
+		return
+	}
+	bucketID, err := c.getBucketIdForDBRP(db, rp)
+	if err != nil {
+		color.Red("Unable to match DBRP to BucketID: %v", err)
+		return
+	}
+	writeReq := c.PostWrite(context.Background()).
+		Bucket(bucketID).
+		Precision(api.WritePrecision(c.Precision)).
+		ContentEncoding("gzip").
+		Body(buf.Bytes())
+	if c.OrgID != "" {
+		writeReq = writeReq.Org(c.OrgID)
+	} else if c.OrgName != "" {
+		writeReq = writeReq.Org(c.OrgName)
+	} else {
+		writeReq = writeReq.Org(c.ActiveConfig.Org)
+	}
+	if err := writeReq.Execute(); err != nil {
+		color.Red("ERR: %v", err)
+		if c.Database == "" {
+			fmt.Println("Note: error may be due to not setting a database or retention policy.")
+			fmt.Println(`Please set a database with the command "use <database>"`)
+			return
+		}
+	}
+}
+
+// Reverse search for a bucket from db & rp
+func (c Client) getBucketIdForDBRP(db string, rp string) (string, error) {
+	dbrpsReq := c.GetDBRPs(context.Background())
+	if c.OrgID != "" {
+		dbrpsReq = dbrpsReq.OrgID(c.OrgID)
+	}
+	if c.OrgName != "" {
+		dbrpsReq = dbrpsReq.Org(c.OrgName)
+	}
+	if c.OrgID == "" && c.OrgName == "" {
+		dbrpsReq = dbrpsReq.Org(c.ActiveConfig.Org)
+	}
```

Comment:
This code looks familiar... any way we could reuse the code in `clients/params.go#GetOrgID()` maybe with a new interface instead of directly taking `api.OrganizationsApi` (if it seems like a simple enough change to make)